rct229/rulesets/ashrae9012019/ruleset_functions/get_aggregated_zone_hvac_fan_operating_schedule.py

```diff
@@ -0,0 +1,56 @@
+import pandas as pd
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_list_hvac_systems_associated_with_zone import (
+    get_list_hvac_systems_associated_with_zone,
+)
+from rct229.utils.assertions import assert_, getattr_
+from rct229.utils.jsonpath_utils import find_one
+
+
+def get_aggregated_zone_hvac_fan_operating_schedule(rmd, zone_id):
+    """
+     This function loops through all of the HVAC system fan operating schedules associated with a specific zone and
+     creates an aggregated fan operating schedule for the zone. More specifically, if any of the fan operating schedules
+     associated with any of the hvac systems serving the zone have a 1 for a particular hour of the year then the aggregated schedule will equal 1
+     for that particular hour of the year. The function will check this for each hour of the year and return an 8760 aggregated fan operating schedule.
+
+    Parameters
+    ----------
+    rmd: dict A zone id associated with the RMR for which to create the aggregated fan operating schedule as described above.
+    zone_id: str A zone id associated with the RMR for which to create the aggregated fan operating schedule as described above.
+
+    Returns
+    ----------
+    aggregated_zone_hvac_fan_operating_schedule: 8760 aggregated fan operating schedule for the zone.
+    """
+
+    schedules = []
+    for hvac_id in get_list_hvac_systems_associated_with_zone(rmd, zone_id):
+        fan_sys = find_one(
+            f'$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*][?(@.id = "{hvac_id}")].fan_system',
+            rmd,
+        )
+        fan_sys_operating_sch = getattr_(fan_sys, "fan_system", "operating_schedule")
+        fan_sys_sch = find_one(
+            f'$.schedules[*][?(@.id = "{fan_sys_operating_sch}")]', rmd
+        )
```

Comment:
I do not think this is correct
"If the schedule is not present" means "operating_schedule" data is missing. Not because the data is not exist in the Schedule data group.
So the implementation here should be:
```python
fan_sys_operating_sch = fan_sys.get("operating_schedule")
if fan_sys_operating_sch:
    schedules.append(getattr_(find_one(
            f'$.schedules[*][?(@.id = "{fan_sys_operating_sch}")]', rmd
        ), "schedules", "hourly_values"
else:
    schedules.append([1] * 8760)
```

In the above code, we get `fan_sys_operating_sch` from the `fan_system`. The `find_one` should guaranteed return a schedule data because RCT has non-schema validation that checks all the associations.

Reply:
Great point! Thank you for this correction.

Based on your code, I made a small update - I considered a scenario when there is no returned schedule from the code shown below. This was tested with the first unit test (```test__get_aggregated_zone_hvac_fan_operating_schedule__no_exist_schedule```).
```Python
find_one(f'$.schedules[*][?(@.id = "{fan_sys_operating_sch}")]', rmd)
```